js/damage.js

```diff
@@ -305,7 +307,10 @@ function getDamageResult(attacker, defender, move, field) {
         description.weather = field.weather;
     } else if (gen >= 6 && move.name === "Knock Off" && !(defender.item === "" ||
             (defender.name === "Giratina-O" && defender.item === "Griseous Orb") ||
-            (defender.name.indexOf("Arceus") !== -1 && defender.item.indexOf("Plate") !== -1))) {
+            (defender.name.indexOf("Arceus") !== -1 && defender.item.indexOf("Plate") !== -1) ||
+            (defender.name.indexOf("Genesect") !== -1 && defender.item.indexOf("Drive") !== -1) ||
+            (defender.name.indexOf("Groudon") && defender.item === "Red Orb") ||
+            (defender.name.indexOf("Kyogre") && defender.item === "Blue Orb"))) {
```

Comment:
You still have to check the `indexOf()` value like the others, otherwise this will work for every Pokemon _except_ Groudon and Kyogre.